sklearnserver/tests/test_rock.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+#
+# Tests for required artifacts to be present in the ROCK image.
+#
+
+from pathlib import Path
+
+import os
+import logging
+import random
+import pytest
+import string
+import subprocess
+import yaml
+from pytest_operator.plugin import OpsTest
+
+def read_rock_info():
+    ROCKCRAFT = yaml.safe_load(Path("rockcraft.yaml").read_text())
+    name = ROCKCRAFT["name"]
+    version = ROCKCRAFT["version"]
+    arch = list(ROCKCRAFT["platforms"].keys())[0]
+    return f"{name}_{version}_{arch}:{version}"
+
+@pytest.fixture()
+def rock_test_env():
+    """Yields a temporary directory and random docker container name, then cleans them up after."""
+    container_name = "".join([str(i) for i in random.choices(string.ascii_lowercase, k=8)])
+    yield container_name
+
+    try:
+        subprocess.run(["docker", "rm", container_name])
+    except Exception:
+        pass
+
+@pytest.mark.abort_on_fail
+def test_rock(ops_test: OpsTest, rock_test_env):
+    """Test rock."""
+    container_name = rock_test_env
+    LOCAL_ROCK_IMAGE = read_rock_info()
+
+    # verify that all artifacts are in correct locations
+    subprocess.run(["docker", "run", LOCAL_ROCK_IMAGE, "exec", "ls", "-la", "/microservice/SKLearnServer.py"], check=True)
```

Comment:
We camn put read_rock_info() in chisme.
rock_tesn_env() is different, since we do not take `tmp_dir`
And test_rock() is specific for sklearnserver and cannot be re-used.
Proposal:
read_rock_info() into chisme
rock_tesn_env(tmp_dir) into chisme
rock_tesn_env() into chisme

Reply:
I like it.

Can we also encapsulate the docker subprocess call into chisme as `assert_file_in_docker_image()`?  That way we don't duplicate the docker syntax in every rock test.  And maybe a convenience function of `assert_files_in_docker_image()` that accepts a list of files

Reply:
oh wait, this method only works when the container has `ls`.  This is going to have the same problems as seldon-core-operator's [test_rock.py](https://github.com/canonical/seldonio-rocks/blob/main/seldon-core-operator/tests/test_rock.py) - if we move to bare, it breaks the tests.  Why not do this like you did in [this test](https://github.com/canonical/seldonio-rocks/blob/main/seldon-core-operator/tests/test_rock.py#L39)?

Reply:
Created a test class in chisme. Will re-do this.
Fixtures will have to stay with tests though.